### PR TITLE
Update GCP Spring starter

### PR DESCRIPTION
**Pull request by mfcrocker**
# What and why?
There was a security advisory on the GCP Spring Datastore starter. This has been disowned by Spring and taken up by Google themselves in the last couple of years, so this PR updates to reflect that

# How to test?
Smoke test banners

# Trello
RAS-314

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -61,7 +67,7 @@
         <dependency>
             <groupId>com.fasterxml.jackson.core</groupId>
             <artifactId>jackson-databind</artifactId>
-            <version>2.13.4.2</version>
+            <version>2.15.1</version>
```
Comment (pricem14pc) [inline pom.xml]: This is the most recent `jackson-databind` dependency that supports the `logstash-logback-encoder`. The most recent versions at `2.15.X` does not have the `StreamConstraintsException` class

File: src/main/resources/logback.xml [hunk 2]
```diff
@@ -9,15 +9,7 @@
         <thread>[ignore]</thread>
         <levelValue>[ignore]</levelValue>
       </fieldNames>
-      <providers>
-        <globalCustomFields>
-          <customFields>{"service":"banner-api"}</customFields>
-        </globalCustomFields>
-        <logLevel>
```
Comment (pricem14pc) [inline src/main/resources/logback.xml:16]: The Logback config has changed. The `logLevel` is reported as 'ignored' by Logback and doesn't need to be configured here (so removed it). When testing in a local env, the log level and severity seems to be included in the structured log by default

File: src/main/resources/logback.xml [hunk 3]
```diff
@@ -9,15 +9,7 @@
...
-          <fieldName>level</fieldName>
-          <fieldName>severity</fieldName>
-        </logLevel>
-      </providers>
+      <customFields>{"service":"banner-api"}</customFields>
```
Comment (pricem14pc) [inline src/main/resources/logback.xml:12]: The config for `customField` has changed in this version of `LogstashEncoder` and no longer needs to be nested in a `providers.globalCustomFields`

**Review by pricem14pc — COMMENTED** [review]
Tested in a local development environment, unit and integration tests passing locally, local docker build successful, testing of functionality tested successful (creating and viewing banner messages).

**Review by pricem14pc — APPROVED** [review]
Also successfully tested in a GCP Cloud environment, logging and functionality all working as expected following Google library and Logback dependency and config changes.

File: pom.xml [hunk 4]
```diff
@@ -52,25 +67,19 @@
         <dependency>
             <groupId>com.fasterxml.jackson.core</groupId>
             <artifactId>jackson-databind</artifactId>
-            <version>2.13.4.2</version>
+            <version>2.14.2</version>
```
Comment (pricem14pc) [inline pom.xml:70]: This is the most recent `jackson-databind` dependency that supports the `logstash-logback-encoder`. The most recent versions at `2.15.X` does not have the `StreamConstraintsException` class

**Review by SteveScorfield — APPROVED** [review]
Acceptances tests ran and all passed. Code looks fine. Approved

**Pull request merged**